apps/storefront/src/pages/Address/components/AddressItemCard.tsx

```diff
@@ -16,13 +17,12 @@ import { AddressItemType } from '../../../types/address';
 
 export interface OrderItemCardProps {
   item: AddressItemType;
-  onEdit: (data: AddressItemType) => void;
-  onDelete: (data: AddressItemType) => void;
-  onSetDefault: (data: AddressItemType) => void;
-  editPermission: boolean;
-  isBCPermission: boolean;
-  updateActionsPermission: boolean;
-  deleteActionsPermission: boolean;
+  canEdit: boolean;
+  onEdit: () => void;
```

Comment:
I go back and forth on this kind of setup or just a 

```tsx
export interface OrderItemCardProps {
  // etc etc
  onEdit?: () => void;
}

// etc etc

{onEdit && (
  <IconButton aria-label="edit" size="small" onClick={onEdit}>
    <EditIcon fontSize="inherit" />
  </IconButton>
)}
```

🍹

Reply:
I changed it to this pattern, I feel a bit awkward about it though. Not sure if it is the best, but worth a try.